sql/expression/comparison.go

```diff
@@ -68,6 +68,11 @@ type comparison struct {
 }
 
 func newComparison(left, right sql.Expression) comparison {
+	// TODO: somewhat hacky way to disable rounding for comparisons
+	setArithmeticOps(left, -1)
+	setArithmeticOps(right, -1)
+	setDivs(left, -1)
+	setDivs(right, -1)
```

Comment:
You might consider a helper function like `disableRounding(e sql.Expression)` that would call setArithmeticOps and setDivs, then it would be pretty explicit what you're doing when you use that functionality and I think you could drop the TODO note, since that helper function name would make the intent clear.